doc/function_types.md

```diff
@@ -1144,3 +1144,15 @@ including ensmallen:
 This can be useful for situations where you know that the checks should be
 ignored.  However, be aware that the code may fail to compile and give more
 confusing and difficult error messages!
+
+
+## Multi-objective functions
+
+A multi-objective function is a set of functions or objectives that are
+simultaneously optimised. For this variety of problems, a class/struct
+with a speicfic API must be used which is illustrated in the file
+`include/ensmallen_bits/problems/schaffer_function_n1.hpp` for the Schaffer N.1
+function.
```

Comment:
Can you add the details here, in the same way that they exist for the other function types?  e.g., a clear description of what a multi-objective function is (mathematically) and what it means to optimize them simultaneously, plus a full description of the API and an example implementation of a function.  I know it's a lot to write, but I think it's really important that we get the documentation right.  Unclear documentation (or documentation that forces users to go digging in the source) is probably going to mean that users don't actually use the functionality, and I want to make sure the multi-objective optimizers you (and others) are implementing get widespread usage. :smile:

Reply:
Sure, I'll get to it right away. :)